libproxy/proxy.cpp

```diff
@@ -56,398 +37,57 @@ class proxy_factory {
-		method = response;
-	else {
-		method = response.substr(0, response.find_first_of(" \t"));
-		server = response.substr(response.find_first_of(" \t") + 1);
-	}
-
-	// Insert our url value
-	if (istringcmp(method, "proxy") && url::is_valid("http://" + server))
-		retval.insert(retval.begin(), string("http://") + server);
-	else if (istringcmp(method, "socks") && url::is_valid("http://" + server))
-		retval.insert(retval.begin(), string("socks://") + server);
-	else if (istringcmp(method, "socks4") && url::is_valid("http://" + server))
-		retval.insert(retval.begin(), string("socks4://") + server);
-	else if (istringcmp(method, "socks4a") && url::is_valid("http://" + server))
-		retval.insert(retval.begin(), string("socks4a://") + server);
-	else if (istringcmp(method, "socks5") && url::is_valid("http://" + server))
-		retval.insert(retval.begin(), string("socks5://") + server);
-	else if (istringcmp(method, "direct"))
-		retval.insert(retval.begin(), string("direct://"));
-}
-
 proxy_factory::proxy_factory() {
-	const char *module_dir;
-
-#ifdef WIN32
-	this->mutex = CreateMutex(NULL, false, NULL);
-	WSADATA wsadata;
-	WORD vers = MAKEWORD(2, 2);
-	WSAStartup(vers, &wsadata);
-#else
-	pthread_mutex_init(&this->mutex, NULL);
-#endif
-
-	lock();
-
-	this->pac    = NULL;
-	this->pacurl = NULL;
-	this->wpad   = false;
 
-	// Register our types
-	this->mm.register_type<config_extension>();
-	this->mm.register_type<ignore_extension>();
-	this->mm.register_type<network_extension>();
-	this->mm.register_type<pacrunner_extension>();
-	this->mm.register_type<wpad_extension>();
-
-	// Load builtin modules
-	for (int i=0 ; builtin_modules[i] ; i++)
-		this->mm.load_builtin(builtin_modules[i]);
-
-	// Load all modules
-	module_dir = getenv("PX_MODULE_PATH");
-	if (!module_dir)
-		module_dir = MODULEDIR;
-	this->mm.load_dir(module_dir);
-	this->mm.load_dir(module_dir, false);
-
-	this->debug  = (getenv("_PX_DEBUG") != NULL);
-
-	unlock();
+  this->proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
+                                         G_DBUS_PROXY_FLAGS_NONE,
+                                         NULL, /* GDBusInterfaceInfo */
+                                         "org.libproxy.proxy",
+                                         "/org/libproxy/proxy",
+                                         "org.libproxy.proxy",
+                                         NULL, /* GCancellable */
+                                         this->error);
 }
 
 proxy_factory::~proxy_factory() {
-	lock();
-
-	if (this->pac) delete[] this->pac;
-	if (this->pacurl) delete this->pacurl;
-	
-	unlock();
 
-#ifdef WIN32
-	WSACleanup();
-	ReleaseMutex(this->mutex);
-#else
-	pthread_mutex_destroy(&this->mutex);
-#endif
 }
 
 
 vector<string> proxy_factory::get_proxies(const string &realurl) {
 	vector<string>             response;
 
-	// Check to make sure our url is valid
-	if (!url::is_valid(realurl))
-		goto do_return;
 
-	lock();
+GVariant* result;
```

Comment:
Maybe GitHub is lying to me, but it looks like there's no indentation here at all. I would double-check this. Indentation looks weird throughout this file tbh.